compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -957,6 +963,63 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       case ctor: MethodSymbol if ctor.isPrimaryConstructor => ctor
     }
 
+  private def needsFixOrCleanup(template: Template): Boolean = {
+    val body = template.body
+    if (body == null) {
+      false
+    } else {
+      body.exists { member =>
+        val sym = member.symbol
+        if (sym != null) sym.hasAnnotation(WasPublicBeforeTyperClass)
+        else false
+      }
+    }
+  }
+
+  private def fixPublicBeforeTyper(template: Template): Template = {
+    val body = template.body
+
+    def needsCleanUp(vddef: ValOrDefDef): Boolean =
+      vddef.symbol.hasAnnotation(WasPublicBeforeTyperClass)
+
+    val isInJSAny = isJSAny(template.symbol.owner)
+
+    val needsFix: DefDef => Boolean = {
+      if (isInJSAny) {
+        new Function1[DefDef, Boolean] {
+          val accessorsNeedFix = body.collect {
+            case vdef: ValDef if needsCleanUp(vdef) => vdef.symbol
+          }.toSet
+          def apply(ddef: DefDef): Boolean = {
+            val sym = ddef.symbol
+            sym.isPrivate &&
+            (sym.isAccessor && accessorsNeedFix(ddef.symbol.accessed)) ||
```

Comment:
That does not work because when I reach the accessor method the annotation has been erased. Note that the members are processed in order and that variable definitions come before their accessors. Alternatively I could do the transformation in two passes, first handle the methods and then the variables.